### DISCUSSION TITLE
Wonky rotary encoder behavior

### DISCUSSION THREAD

**Discussion by scottmbaker** [keirf/flashfloppy#521] [General]
Not filing this as a bug because I think the problem is mine, not with the software, but I'll be damned if I can figure out what it is.

I have a gotek SFR1M44-U100, the older version with the six-pin header up front, not the newer one with the 4-pin. I've attached a plain ordinary rotary encoder to it. The encoder works fine on my bench, connected to my bench 5V supply. As soon as I put it in the computer, the encoder will not work. A few times it was intermittent, but lately it does not work at all. Pull it from the computer and put it back on the bench and the encoder works fine again. I've repeated this in and out of the computer a half dozen times. All other functions (prev and next buttons, oled, etc) work fine. I have verified that all three wires to the encoder are good. It's a bare encoder, no pcboard or pullups.

I have tried `rotary = full`, `rotary = v2`, and `rotary = half` in the ff.cfg with not success (`rotary = half` will mess up behavior on the bench, as expected). 

Anyone have a suggestion?

Thanks,
Scott

**Comment by keirf** [comment]
Have you tried another rotary? That's the first thing to try, since they're so cheap. Also try with FF v2 as the rotary handling changed in v3 and became stricter to register a movement.

**Reply by scottmbaker** [reply]
After spending an hour tonight with an oscilloscope on the encoder, with a second encoder, various stages of attached to the computer and unattached, with or without the computer's second FDD, holding the case in various orientations, etc., I have solved the problem... and I feel like an idiot. It was a simple piece of metal, which as soon as you assemble the case snaps up into the bottom of the Gotek and shorts the encoder header pins.

![qx10gotek](https://user-images.githubusercontent.com/10369013/130715141-3411927c-f350-4a2f-a200-282512b51376.JPG)

It's obvious. I can't believe I didn't see it.
